### PR TITLE
Add CalculateNumberAssetCommand to OpenBots.Commands.Asset

### PR DESCRIPTION
**Pull request by openbots-nc**
- Create CalculateNumberAssetCommand.cs in OpenBots.Commands.Asset project
- Add necessary methods in AssetMethods and image in UIImage
- Create appropriate unit tests for new command in CalculateNumberAssetCommandTests.cs under OpenBots.Command.Asset.Tests project

### CODE REVIEW

File: OpenBots.Commands/OpenBots.Commands.Asset/CalculateNumberAssetCommand.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    [Serializable]
+    [Category("Asset Commands")]
+    [Description("This command increments, decrements, adds, or subtracts an Asset in OpenBots Server.")]
+    public class CalculateNumberAssetCommand : ScriptCommand
+    {
+        [Required]
+        [DisplayName("Number Asset Name")]
+        [Description("Enter the name of the Asset.")]
+        [SampleUsage("Name || {vCredentialName}")]
+        [Remarks("This command will throw an exception if an asset of the wrong type is used")]
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Asset/CalculateNumberAssetCommand.cs:27]:
```suggestion
        [Remarks("This command will throw an exception if an asset of the wrong type is used.")]
```

File: OpenBots.Commands/OpenBots.Commands.Asset/CalculateNumberAssetCommand.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                ctrl.Visible = false;
+            RenderedControls.AddRange(_assetActionValueControls);
+
+            return RenderedControls;
+        }
+
+        public override string GetDisplayValue()
+        {
+            if (v_AssetActionType == "Increment")
+                return base.GetDisplayValue() + $" (Increment) ['{v_AssetName}']";
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Asset/CalculateNumberAssetCommand.cs:112]:
```suggestion
                return base.GetDisplayValue() + $" [Increment '{v_AssetName}']";
```

Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.Asset/CalculateNumberAssetCommand.cs:112]: square brackets should surround the entire display message. Please use other commands as a reference.

**Review by openbots-ff — CHANGES_REQUESTED** [review]
Looks good. Please add tests for this command as well

File: OpenBots.Commands/OpenBots.Commands.Asset/CalculateNumberAssetCommand.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        {
+            var engine = (AutomationEngineInstance)sender;
+            var vAssetName = v_AssetName.ConvertUserVariableToString(engine);
+            var vAssetActionValue = v_AssetActionValue.ConvertUserVariableToString(engine);
+
+            var client = AuthMethods.GetAuthToken();
+            var asset = AssetMethods.GetAsset(client, $"name eq '{vAssetName}' and type eq 'Number'");
+
+            if (asset == null)
+                throw new Exception($"No Asset was found for '{vAssetName}' and type 'Number'");
```
Comment (openbots-su) [inline OpenBots.Commands/OpenBots.Commands.Asset/CalculateNumberAssetCommand.cs:73]: Change this to an ObjectNotFoundException please.
https://docs.microsoft.com/en-us/dotnet/api/system.data.objectnotfoundexception?view=netframework-4.8

Comment (openbots-nc) [reply OpenBots.Commands/OpenBots.Commands.Asset/CalculateNumberAssetCommand.cs]: Changed to DataException to stay consistennt with Get Asset and Update Asset Commands

File: OpenBots.Commands/OpenBots.Commands.Asset/CalculateNumberAssetCommand.cs [hunk 4]
```diff
@@ -109,7 +110,7 @@ public override List<Control> Render(IfrmCommandEditor editor, ICommandControls
         public override string GetDisplayValue()
         {
             if (v_AssetActionType == "Increment")
-                return base.GetDisplayValue() + $" (Increment) ['{v_AssetName}']";
+                return base.GetDisplayValue() + $" [Increment '{v_AssetName}']";
             else if (v_AssetActionType == "Decrement")
                 return base.GetDisplayValue() + $" (Decrement) ['{v_AssetName}']";
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.Asset/CalculateNumberAssetCommand.cs:115]: You forgot to update the other ones.

**Review by openbots-ff — CHANGES_REQUESTED** [review]

**Review by openbots-ff — DISMISSED** [review]

**Review by openbots-ff — APPROVED** [review]

**Review by openbots-ff — APPROVED** [review]

**Review by openbots-su — APPROVED** [review]

**Pull request merged**